### PR TITLE
Add new snap metadata

### PR DESCRIPTION
**Pull request by danroc**
The added metadata will be visible to users when they're adding a new snap account. All the newly introduced fields are optional, as they might not be necessary for other types of snaps.

Closes: MetaMask/accounts-planning/issues/21

### CODE REVIEW

**Comment by Montoya** [comment]
I've looked at this a bunch of times and I think I like it. It would be helpful to start using this. I have a feeling we will eventually decide to change this approach and do something different with a "v2" but we can cross that bridge when we get there. I'm curious to hear @FrederikBolding's thoughts as well.

File: src/index.ts [hunk 1]
```diff
@@ -21,6 +22,13 @@ export const VerifiedSnapStruct = object({
   id: string(),
   metadata: object({
     name: string(),
+    type: optional(enums(['account'])),
+    author: optional(string()),
+    website: optional(string()),
+    summary: optional(string()),
+    description: optional(string()),
+    reports: optional(array(string())),
```
Comment (FrederikBolding) [inline src/index.ts:30]:
```suggestion
    audits: optional(array(string())),
```

File: src/index.ts [hunk 2]
```diff
@@ -21,6 +22,13 @@ export const VerifiedSnapStruct = object({
   id: string(),
   metadata: object({
     name: string(),
+    type: optional(enums(['account'])),
+    author: optional(string()),
+    website: optional(string()),
+    summary: optional(string()),
```
Comment (FrederikBolding) [inline src/index.ts:28]: What's the difference between a summary and a description here?

Comment (danroc) [reply src/index.ts:28]: `summary` is a short summary that we will display in the cards view, `description` is a (longer) description that we show when a snap is selected from the list.

It's the same meaning as the in the [OpenAPI specs](https://spec.openapis.org/oas/v3.1.0#fixed-fields-0).

**Review by FrederikBolding — APPROVED** [review]
LGTM. Please hold off merging until @Mrtenz also looks at this

**Review by Mrtenz — COMMENTED** [review]
A support URL could be useful?

**Comment by danroc** [comment]
> A support URL could be useful?

Could be indeed. @AlexJupiter, WDYT?

File: src/index.ts [hunk 3]
```diff
@@ -21,6 +22,13 @@ export const VerifiedSnapStruct = object({
...
+    description: optional(string()),
+    audits: optional(array(string())),
+    tags: optional(array(string())),
```
Comment (FrederikBolding) [inline src/index.ts:31]:
```suggestion
    tags: optional(array(string())),
    sourceCodeUrl: optional(string()),
    supportUrl: optional(string()),
```

**Comment by AlexJupiter** [comment]
> > A support URL could be useful?
> 
> Could be indeed. @AlexJupiter, WDYT?

Yes, sure, this seems like a good addition.

File: src/index.ts (hunk 3 above)
Comment (danroc) [reply src/index.ts:31]: Should we keep `website` or apply the same pattern and rename it to `websiteUrl`?

Comment (Mrtenz) [reply src/index.ts:31]: Could drop the `-url`?

Comment (danroc) [reply src/index.ts:31]: @Mrtenz only for `website` or for all fields (`sourceCode` and `support`)?

Comment (Mrtenz) [reply src/index.ts:31]: All fields?

Comment (FrederikBolding) [reply src/index.ts:31]: I'm fine with nixing the url suffix

**Review by Mrtenz — APPROVED** [review]

File: src/index.ts (hunk 3 above)
Comment (danroc) [reply src/index.ts:31]: All right, nixing the `Url` suffix after checking with @FrederikBolding on Slack :)

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**